### PR TITLE
Feature/support custom aws endpoint

### PR DESCRIPTION
**Pull request by graysonchen**

### CODE REVIEW

File: app/views/camaleon_cms/admin/settings/_file_system_settings.html.erb [hunk 1]
```diff
@@ -20,13 +20,13 @@
             <%= label_tag t('camaleon_cms.admin.settings.filesystem_region', default: 'filesystem_region') %> <small>(*)</small><br>
             <%= text_field :options, :filesystem_region, :class => "form-control required", value: @site.get_option("filesystem_region"), placeholder: 'us-west-2' %>
         </div>
-        <!--<div class="form-group">
-            <%#= label_tag t('camaleon_cms.admin.settings.filesystem_s3_endpoint', default: 'filesystem_s3_endpoint') %> <small>(*)</small><br>
-            <%#= text_field :options, :filesystem_s3_endpoint, :class => "form-control required", value: @site.get_option("filesystem_s3_endpoint"), placeholder: 's3-us-west-2.amazonaws.com' %>
-        </div>-->
+        <div class="form-group">
```
Comment (texpert) [inline app/views/camaleon_cms/admin/settings/_file_system_settings.html.erb:23]: Thanks for the PR, @graysonchen !

Let's ask first @owen2345 why the code has been commented from the very creation? Are there any obstacles to enable it?

Comment (owen2345) [reply app/views/camaleon_cms/admin/settings/_file_system_settings.html.erb:23]: AFAIR, the implementation was not complete and thus I preserved the HTML part commented.

**Review by owen2345 — APPROVED** [review]
LGTM

**Comment by graysonchen** [comment]
hey @texpert is it ready to merge it? @owen2345 Thank you feedback.

File: app/uploaders/camaleon_cms_aws_uploader.rb [hunk 2]
```diff
@@ -140,8 +141,13 @@ def delete_file(key)
...
       Aws.config.update({ region: @aws_region, credentials: Aws::Credentials.new(@aws_akey, @aws_asecret) })
-      s3 = Aws::S3::Resource.new
+      s3 = Aws::S3::Resource.new(resource_parameters)
       s3.bucket(@aws_bucket)
     }.call
   end
+
+  def resource_parameters
+    return {} if @aws_endpoint.blank?
+    { endpoint: @aws_endpoint }
```
Comment (graysonchen) [inline app/uploaders/camaleon_cms_aws_uploader.rb:151]:
```suggestion

    { endpoint: @aws_endpoint }
```

**Comment by texpert** [comment]
OK, I've fixed the Rubocop issue, and the remaining failure is caused by `font-awesome-rails` and `non-digest-assets` not yet supporting the Rails 8 version. So, merging it!

Thanks again, @graysonchen!

**Pull request merged**